### ISSUE TITLE
Error when creating cucumber.json TypeError: Cannot read property 'status' of undefined

### ISSUE THREAD

**Issue by inavac182** [cucumber/cucumber-js#1204]
Hello guys,

I'm running into a weird error when running my test with nightwatch and cucumber in parallel. I'm seeing this error:

```
TypeError: Cannot read property 'status' of undefined
11:17:33     at SummaryFormatter.isTestCaseFailure (......node_modules/cucumber/lib/formatter/summary_formatter.js:25:106)
11:17:33     at _lodash.default.each.testCase (......node_modules/cucumber/lib/formatter/summary_formatter.js:37:16)
11:17:33     at ......node_modules/lodash/lodash.js:4911:15
11:17:33     at baseForOwn (......node_modules/lodash/lodash.js:2996:24)
11:17:33     at ......node_modules/lodash/lodash.js:4880:18
11:17:33     at Function.forEach (......node_modules/lodash/lodash.js:9344:14)
11:17:33     at SummaryFormatter.logSummary (......node_modules/cucumber/lib/formatter/summary_formatter.js:36:21)
11:17:33     at EventEmitter.PrettyFormatter.options.eventBroadcaster.on (......node_modules/cucumber-pretty/index.js:116:17)
11:17:33     at emitOne (events.js:121:20)
11:17:33     at EventEmitter.emit (events.js:211:7)
11:17:33     at Master.onSlaveClose (......node_modules/cucumber/lib/runtime/parallel/master.js:96:29)
11:17:33     at ChildProcess.slave.process.on (......node_modules/cucumber/lib/runtime/parallel/master.js:83:12)
11:17:33     at emitTwo (events.js:126:13)
11:17:33     at ChildProcess.emit (events.js:214:7)
11:17:33     at maybeClose (internal/child_process.js:925:16)
11:17:33     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
```

Looks like when there are certain errors during the execution of the tests, somehow this is generating this weird error that is like flaky because it happen sometimes.

In this particular case, I'm seeing these errors in the console: 

```
11:17:08 Error while running .getAllWindowHandles() protocol action: An unknown server-side error occurred while processing the command.
```

And:

```
Error while running .getScreenshot() protocol action: An unknown server-side error occurred while processing the command.
11:17:19 
11:17:19    Error: Creating screenshot was not successful. Response was:
11:17:19   { status: -1,
11:17:19     state: 'error',
11:17:19     value: 
11:17:19      { message: 'An unknown server-side error occurred while processing the command.',
11:17:19        error: [] },
11:17:19     errorStatus: 13,
11:17:19     error: 'An unknown server-side error occurred while processing the command.',
11:17:19     httpStatusCode: 200 }
11:17:19    { status: -1,
11:17:19      state: 'error',
11:17:19      value: 
11:17:19       { message: 'An unknown server-side error occurred while processing the command.',
11:17:19         error: [] },
11:17:19      errorStatus: 13,
11:17:19      error: 'An unknown server-side error occurred while processing the command.',
11:17:19      httpStatusCode: 200 }
11:17:19        at <anonymous>
11:17:19        at process._tickCallback (internal/process/next_tick.js:188:7)
```

As I'm running in parallel the console is a bit messy so, it's hard to identify which of this error is creating the `status of undefined` in the reporter.

But I have seen this error also happen when an assertion inside of a .execute() fails or when we try to getText of an element that is not present in the DOM, again as this error is flaky and the console is a bit messy I could be wrong with this scenarios. 

So, probably in this case it would be better that in the reporter we could have a safe check, to report this scenario like failed instead of breaking the entire execution with `TypeError: Cannot read property 'status' of undefined`. That way we can still be getting the report and identify that there is an issue somewhere else that we need to fix.

As this error is coming from: `node_modules/cucumber/lib/formatter/summary_formatter.js` I created this issue here in the cucumber-js repo.

We are using:
cucumber: 5.1.0
nightwatch-api: 2.2.1
nightwatch: 1.1.8

➜  ~ node -v
v10.15.3

**Comment by inavac182** [comment]
I'm getting this error also when the global timeout is reached:

```
Error: function timed out, ensure the promise resolves within 60000 milliseconds
13:54:53           at Timeout._time.default.setTimeout [as _onTimeout] (...../node_modules/cucumber/lib/user_code_runner.js:81:20)
13:54:53           at ontimeout (timers.js:482:11)
13:54:53           at tryOnTimeout (timers.js:317:5)
13:54:53           at Timer.listOnTimeout (timers.js:277:5)

13:54:53 
....../node_modules/cucumber/lib/formatter/summary_formatter.js:25
13:54:53     return _lodash.default.includes([_status.default.AMBIGUOUS, _status.default.FAILED], testCase.result.status);
13:54:53                                                                                                          ^
13:54:53 
13:54:53 
TypeError: Cannot read property 'status' of undefined
13:54:53     at SummaryFormatter.isTestCaseFailure (....../node_modules/cucumber/lib/formatter/summary_formatter.js:25:106)
13:54:53     at _lodash.default.each.testCase (....../node_modules/cucumber/lib/formatter/summary_formatter.js:37:16)
13:54:53     at ....../node_modules/lodash/lodash.js:4911:15
13:54:53     at baseForOwn (....../node_modules/lodash/lodash.js:2996:24)
13:54:53     at ....../node_modules/lodash/lodash.js:4880:18
13:54:53     at Function.forEach (....../node_modules/lodash/lodash.js:9344:14)
13:54:53     at SummaryFormatter.logSummary (....../node_modules/cucumber/lib/formatter/summary_formatter.js:36:21)
13:54:53     at EventEmitter.PrettyFormatter.options.eventBroadcaster.on (....../node_modules/cucumber-pretty/index.js:116:17)
13:54:53     at emitOne (events.js:121:20)
13:54:53     at EventEmitter.emit (events.js:211:7)
13:54:53     at Master.onSlaveClose (......//node_modules/cucumber/lib/runtime/parallel/master.js:96:29)
13:54:53     at ChildProcess.slave.process.on (....../node_modules/cucumber/lib/runtime/parallel/master.js:83:12)
13:54:53     at emitTwo (events.js:126:13)
13:54:53     at ChildProcess.emit (events.js:214:7)
13:54:53     at maybeClose (internal/child_process.js:925:16)
13:54:53     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
13:54:53 npm ERR! code ELIFECYCLE
13:54:53 npm ERR! errno 1
13:54:53 npm ERR! fsr-nightwatch@2.0.4 e2e-test-run: `cucumber-js `
13:54:53 npm ERR! Exit status 1
13:54:53 npm ERR! 
13:54:53 
```

**Comment by inavac182** [comment]
Any update on this?

**Comment by inavac182** [comment]
I was running the tests without parallel flag and got this error in the console:
```
unhandledRejection:
13:18:18    NightwatchAssertError: Error while running "waitForElementPresent" command: Timed out while waiting for element <#fsrDiv> to be present for 10000 milliseconds. - expected "found" but got: "not found"
```

and then the execution was stopped.

```
13:18:18  unhandledRejection:
13:18:18    NightwatchAssertError: Error while running "waitForElementPresent" command: Timed out while waiting for element <#fsrDiv> to be present for 10000 milliseconds. - expected "found" but got: "not found"
13:18:18        at <anonymous>
13:18:18        at process._tickCallback (internal/process/next_tick.js:188:7)
13:18:18 npm ERR! code ELIFECYCLE
13:18:18 npm ERR! errno 1
13:18:18 npm ERR! fsr-nightwatch@2.0.4 e2e-test-run: `cucumber-js `
13:18:18 npm ERR! Exit status 1
13:18:18 npm ERR! 
13:18:18 npm ERR! Failed at the fsr-nightwatch@2.0.4 e2e-test-run script.
13:18:18 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
13:18:18 
13:18:18 npm ERR! A complete log of this run can be found in:
13:18:18 npm ERR!     /data/jenkins/.npm/_logs/2019-05-29T20_18_18_226Z-debug.log
13:18:18 
13:18:18 > node create-html-report.js
13:18:18 
13:18:18 module.js:674
13:18:18     throw err;
13:18:18     ^
13:18:18 
13:18:18 SyntaxError: /reports/cucumber.json: Unexpected end of JSON input
13:18:18     at JSON.parse (<anonymous>)
```

**Comment by inavac182** [comment]
After more investigation I'm closing this issue because I found that the issue is how we are implementing the async and await in our code, we were messing with async/await in callbacks that didn't need it and that was breaking the execution. 

Example what we were doing wrong:

```
await browser.waitForElementVisible('@myImput', 10000, async () => {
            await browser.setValue('@myImput', myValue);
});
```

and this is how it should be (or how it doesn't break):
```
await browser.waitForElementVisible('@myImput', 10000, () => {
          browser.setValue('@myImput', myValue);
});
```
When we do the async/await in callback something happen in the background that blocks the execution and this makes the global timeout to be reached and fire the error with the timeout, that's why we were confused about the real issue.

It was really difficult to find because we had to scope the issue to the scenario that was failing and after that check each of the steps, and the error appear to be in a different line which make it even harder to debug, but at the end we checked all the steps that we were doing and after changing this the test were being executed fine.
